schemes/hack_shallow/cmfmca.F90

```diff
@@ -0,0 +1,978 @@
+    real(kind_phys) :: fqlkm               ! flux total water (top interface)
+    real(kind_phys) :: botflx              ! bottom constituent mixing ratio flux
+    real(kind_phys) :: topflx              ! top constituent mixing ratio flux
+    real(kind_phys) :: efac1               ! ratio q to convectively induced chg (btm lvl)
+    real(kind_phys) :: efac2               ! ratio q to convectively induced chg (mid lvl)
+    real(kind_phys) :: efac3               ! ratio q to convectively induced chg (top lvl)
+    real(kind_phys) :: tb(ncol,pver)       ! working storage for temp (t bar)
+    real(kind_phys) :: shb(ncol,pver)      ! working storage for spec hum (sh bar)
+    real(kind_phys) :: adjfac              ! adjustment factor (relaxation related)
+    real(kind_phys) :: rktp
+    real(kind_phys) :: rk
+    integer         :: i,k                 ! longitude, level indices
+    integer         :: ii                  ! index on "gathered" vectors
+    integer         :: len1                ! vector length of "gathered" vectors
+    integer         :: m                   ! constituent index
+    integer         :: ktp                 ! tmp indx used to track top of convective layer
+
+    ! debug use quantities
+    real(kind_phys) :: rh                  ! relative humidity
+    real(kind_phys) :: es                  ! sat vapor pressure
+    real(kind_phys) :: hsum1               ! moist static energy integral
+    real(kind_phys) :: qsum1               ! total water integral
+    real(kind_phys) :: hsum2               ! final moist static energy integral
+    real(kind_phys) :: qsum2               ! final total water integral
+    real(kind_phys) :: fac                 ! intermediate scratch variable
+    integer         :: n                   ! vertical index     (diagnostics)
+    integer         :: kp                  ! vertical index     (diagnostics)
+    integer         :: kpp                 ! index offset, kp+1 (diagnostics)
+    integer         :: kpm1                ! index offset, kp-1 (diagnostics)
+
+    errmsg = ''
+    errflg = 0
+
+    scheme_name = 'cmfmca'
+
+    !---------------------------------------------------
+    ! Initialize output tendencies
+    !---------------------------------------------------
+    cmfdt   (:ncol,:)     = 0._kind_phys
+    cmfdq   (:ncol,:)     = 0._kind_phys
+    cmfmc_sh(:ncol,:)     = 0._kind_phys
+    cmfdqr  (:ncol,:)     = 0._kind_phys
+    cmfsl   (:ncol,:)     = 0._kind_phys
+    cmflq   (:ncol,:)     = 0._kind_phys
+    qc_sh   (:ncol,:)     = 0._kind_phys
+    rliq_sh (:ncol)       = 0._kind_phys
+
+    ! Check constituents list and locate water vapor index
+    ! (not assumed to be 1)
+    const_check_loop: do m = 1, pcnst
+      call const_props(m)%standard_name(const_standard_name)
+      if (const_standard_name == 'water_vapor_mixing_ratio_wrt_moist_air_and_condensed_water') then
+        const_wv_idx = m
+        exit const_check_loop
+      endif
+    enddo const_check_loop
```

Comment:
I'd recommend using the `ccpp_const_get_idx` in `ccpp_const_utils` that you added. I hope that'll make it easier to find and replace with the framework version eventually...

Reply:
Thanks, updated!